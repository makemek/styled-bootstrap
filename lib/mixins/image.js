import { css } from 'styled-components'

export const img_responsive = (display = 'block') => css`
  display: ${display};
  max-width: 100%;
  height: auto;
`

export const img_retina = (file_1x, file_2x, width_1x, height_1x) => css`
  background-image: url(${file_1x});

  @media
  only screen and (-webkit-min-device-pixel-ratio: 2),
  only screen and (   min--moz-device-pixel-ratio: 2),
  only screen and (     -o-min-device-pixel-ratio: 2/1),
  only screen and (        min-device-pixel-ratio: 2),
  only screen and (                min-resolution: 192dpi),
  only screen and (                min-resolution: 2dppx) {
    background-image: url(${file_2x});
    background-size: ${width_1x} ${height_1x};
  }
`
