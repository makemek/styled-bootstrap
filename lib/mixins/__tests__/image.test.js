import * as image from '../image'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import styled from 'styled-components'
import React from 'react'

// TODO test on media queries
describe('mixins: image', () => {

  describe('img_responsive', () => {
    it('should have display:block by default', () => {
      const Comp = styled.div`
        ${image.img_responsive()}
      `
      const wrapper = shallow(<Comp />)
      expect(wrapper).toHaveStyleRule('display', 'block')
    })

    it('should be able to set display prop', () => {
      const Comp = styled.div`
        ${image.img_responsive('something')}
      `
      const wrapper = shallow(<Comp />)
      expect(wrapper).toHaveStyleRule('display', 'something')
    })
  })

  describe('img_retina', () => {
    it('should set arguments to their expected props', () => {
      const Comp = styled.div`
        ${image.img_retina('foo', 'bar', '1em', '20px')}
      `
      const wrapper = shallow(<Comp />)
      expect(wrapper).toHaveStyleRule('background-image', 'url(foo)')
      expect(wrapper).toMatchSnapshot()
    })
  })
})
