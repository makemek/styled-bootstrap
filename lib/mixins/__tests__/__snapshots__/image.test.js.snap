// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mixins: image img_retina should match snapshot 1`] = `
.c0 {
  background-image: url(/foo);
}

@media only screen and (-webkit-min-device-pixel-ratio:2), only screen and ( min--moz-device-pixel-ratio:2), only screen and ( -o-min-device-pixel-ratio:2/1), only screen and ( min-device-pixel-ratio:2), only screen and ( min-resolution:192dpi), only screen and ( min-resolution:2dppx) {
  .c0 {
    background-image: url(/bar);
    background-size: 1em 20px;
  }
}

<div
  className="c0"
/>
`;
